bluetooth: Classic: HFP: Set eSCO S4 bit if controller supports SC

lylezhu2012 opened the pull request:
The eSCO S4 settings feature should be enabled for CVSD Coding. It is suggested by the HFP specification.

Set the eSCO S4 Settings Supported bit if the Secure connection is supported by controller.

lylezhu2012 committed:
bluetooth: Classic: HFP: Set eSCO S4 bit if controller supports SC

The eSCO S4 settings feature should be enabled for CVSD Coding. It is
suggested by the HFP specification.

Set the eSCO S4 Settings Supported bit if the Secure connection is
supported by controller.

Signed-off-by: Lyle Zhu <lyle.zhu@nxp.com>

jhedberg commented on subsys/bluetooth/host/classic/hfp_ag.c:
```diff
@@ -3865,6 +3865,7 @@ static struct bt_hfp_ag *hfp_ag_create(struct bt_conn *conn)
 
 	/* Set the supported features*/
 	ag->ag_features = BT_HFP_AG_SUPPORTED_FEATURES;
+	ag->ag_features |= BT_FEAT_SC(bt_dev.features) > 0 ? BT_HFP_AG_FEATURE_ESCO_S4 : 0;
```
I believe the return value of the `BT_FEAT_*()` macros is intended to be used as a boolean, so I think the `> 0` can probably be dropped.

lylezhu2012 replied:
Actually, I had that feeling too. But I confirmed again that return value of the `BT_FEAT_*()` is not a boolean type except `BT_FEAT_BREDR()`.

https://github.com/zephyrproject-rtos/zephyr/blob/05899583fdcd1d2a10e0ef3ca619fe5c57367041/include/zephyr/bluetooth/hci_types.h#L143-L149

lylezhu2012 committed:
include: Bluetooth: Change return value of `BT_FEAT_TEST()` to boolean

In current implementation, the return value of `BT_FEAT_TEST()` is not
a boolean type. However, the macro `BT_FEAT_TEST()` is used as a type
with a return type value of boolean.

Change the return type of `BT_FEAT_TEST()` to boolean type to make it
easy to use.

Signed-off-by: Lyle Zhu <lyle.zhu@nxp.com>
